grammar.js

```diff
@@ -537,12 +554,14 @@ module.exports = grammar({
       $.lambda,
       $.primary_expression,
       $.conditional_expression,
-      $.named_expression
+      $.named_expression,
+      $.as_pattern
     ),
 
     primary_expression: $ => choice(
       $.binary_operator,
       $.identifier,
+      alias("match", $.identifier),
```

Comment:
just adding match to "keyword" identifiers did make it work for me to parse `a = [match]` correctly